jina/serve/executors/__init__.py

```diff
@@ -179,14 +179,14 @@ def _init_monitoring(self):
     def _add_requests(self, _requests: Optional[Dict]):
         if not hasattr(self, 'requests'):
             self.requests = {}
-
         if _requests:
             func_names = {f.__name__: e for e, f in self.requests.items()}
             for endpoint, func in _requests.items():
                 # the following line must be `getattr(self.__class__, func)` NOT `getattr(self, func)`
                 # this to ensure we always have `_func` as unbound method
                 if func in func_names:
-                    del self.requests[func_names[func]]
+                    if func_names[func] in self.requests:
```

Comment:
Can we rename `func_name` to `map_func_name_to_request` or something like this ? I find it really confusing to read the code oterhwise. 
PS I know you did not introduce this naming but we can do it in this PR as well.
What do you think @JoanFM ?

Reply:
We will do in another PR if needed, will unblock this for now